hardware/setup/snap_cloud_merge.tcl

```diff
@@ -0,0 +1,159 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#-----------------------------------------------------------
+
+set log_dir               $::env(LOGS_DIR)
+set log_file              $log_dir/snap_cloud_merge.log
+set fpga_part             $::env(FPGACHIP)
+set fpgacard              $::env(FPGACARD)
+set sdram_used            $::env(SDRAM_USED)
+set nvme_used             $::env(NVME_USED)
+set bram_used             $::env(BRAM_USED)
+set static_region_dcp     "snap_static_region_bb.dcp"
+set user_action_dcp       "user_action_synth.dcp"
+
+#timing_lablimit  
+if { [info exists ::env(TIMING_LABLIMIT)] == 1 } {
+  set timing_lablimit [string toupper $::env(TIMING_LABLIMIT)]
+} else {
+  set timing_lablimit "-250"
+}
+
+#checkpoint_dir
+if { [info exists ::env(DCP_ROOT)] == 1 } {
+    set dcp_dir $::env(DCP_ROOT)
+    set file_missing 0
+    if { [file exists $dcp_dir/$static_region_dcp] != 1 } {
+        puts "Error: File \$DCP_ROOT/$static_region_dcp does not exist"
+        set file_missing 1
+    }
+    if { [file exists $dcp_dir/$user_action_dcp] != 1 } {
+        puts "Error: File \$DCP_ROOT/$user_action_dcp does not exist"
+        set file_missing 1
+    }
+    if { $file_missing == 1 } {
+        exit 42
+    }
+} else {
+    puts "Error: For merging user action DCP into snap_static_region DCP the environment variable DCP_ROOT needs to point to the path containing the checkpoints."
+    exit 42
+}
+
+if { [info exists ::env(CLOUD_BUILD_BITFILE)] == 1 } {
+  set cloud_build_bitfile [string toupper $::env(CLOUD_BUILD_BITFILE)]
+} else {
+  set cloud_build_bitfile "FALSE"
+}
+
+#Define widths of each column
+set widthCol1 31
+set widthCol2 23
+set widthCol3 35
+set widthCol4 22
+
+## 
+## create temporary in memory project
+puts [format "%-*s %-*s %-*s"  $widthCol1 "" [expr $widthCol2 + $widthCol3 + 1] "creating in memory project" $widthCol4 "[clock format [clock seconds] -format %H:%M:%S]"]
+create_project -in_memory -part $fpga_part >> $log_file
+
+## 
+## adding static region and user_action checkpoints
+puts [format "%-*s %-*s %-*s %-*s"  $widthCol1 "" $widthCol2 "adding checkpoints" $widthCol3 "" $widthCol4 "[clock format [clock seconds] -format %H:%M:%S]"]
+add_files $dcp_dir/$static_region_dcp >> $log_file
+add_file $dcp_dir/$user_action_dcp >> $log_file
+set_property SCOPED_TO_CELLS {a0/action_w} [get_files $dcp_dir/$user_action_dcp] >> $log_file
+
+## 
+## linking design
+puts [format "%-*s %-*s %-*s %-*s"  $widthCol1 "" $widthCol2 "linking design" $widthCol3 "" $widthCol4 "[clock format [clock seconds] -format %H:%M:%S]"]
+link_design -mode default -reconfig_partitions {user_action} -part $fpga_part -top psl_fpga >> $log_file
+
```

Comment:
Good question - it worked for me without that. Need to do some experiments to see if we should add it.